### DISCUSSION TITLE
Re-applying stash when checking out new branch does not check if initial autostash was successful

### DISCUSSION THREAD

**Discussion by fowie** [gitextensions/gitextensions#11124] [Q&A]
Repro:
Create a stash in your repo (stash0)
Create a new file in your repo.  No modifications.  When you open git extensions commit window you should see one added file, but no modifications.  Do not stage any changes.
Using the repo browser, check out a different branch in your repo.  When the "Checkout branch" dialog box appears, choose "Stash" in the "Local Changes" section, and then check out the branch.  You will see that it runs 
`git.exe stash save `
and the result is
`No local changes to save
Done`
and you get a green check indicating that the operation was successful, but no stash (stash1) was actually created.

Checkout proceeds without any problems, but then you are presented with a dialog box that says "Apply stashed items to working directory again?"  If you choose yes, the original stash0 will be applied to your repo.  

Expected:  No stash should be applied, because no stash was created
Actual:  stash0 is applied to your local copy

**Comment by adlentis** [comment]
I can confirm that this behavior is still present and rendering the autostash function quite useless, I often need to reset and stash manually when switching between branches.

**Comment by gerhardol** [comment]
Include untracked in stash to create a stash in the first scenario

![image](https://github.com/gitextensions/gitextensions/assets/6248932/5b520705-6908-4121-99ff-3cbfd217b446)

Not sure how Git rebase.autostash works here.
This may be a core Git issue

**Comment by vbjay** [comment]
Rebase.autostash reaplies the stash created before rebase work and drops it if no issues. Otherwise keeps stash around and leaves you to solve merge conflicts.

**Comment by RussKie** [comment]
Turning this into a discussion given the age of the issue and a general lack of interest from the community, this looks like a niche issue. 
If git rebase.autostash and the provided "include untracked in stash" don't work, then a pull-request is welcome.
